library/data/loot_table/src/main/java/org/quiltmc/qsl/loot/api/LootJsonParser.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016, 2017, 2018, 2019 FabricMC
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.loot.api;
+
+import java.io.Reader;
+import java.lang.reflect.Field;
+import java.util.stream.Stream;
+
+import com.google.gson.Gson;
+
+import net.minecraft.loot.LootManager;
+import net.minecraft.util.JsonHelper;
+import net.minecraft.util.Lazy;
+
+public final class LootJsonParser {
```

Comment:
This API is also not needed, vanilla's `LootGsons` exposes this functionality. Additionally, the reflection code below is really cursed - I should've used accessors back then.